### PR TITLE
Replace is_password with type to allow more seamless file selection [DEPENDENT ON FORLOOP PLATFORM AND FORLOOP MODULES]

### PR DESCRIPTION
**Pull request by Omastto1**
Replace is_password with type to allow more seamless file selection 
mainly for web - reduce entry + button to entry with 'file' typ only

### CODE REVIEW

**Review by YoungMasterGandalf — APPROVED** [review]

File: pgwidget/pgwidget_core.py [hunk 1]
```diff
@@ -1420,9 +1421,12 @@ def on_unclick(self, glc):
         
 
 class Entry(TextContainerRect):
-    def __init__(self, text, pos, size, color=c.white, is_draggable=True, relative_pos=[0, 0],is_password=False,selection_color=c.red):
-        self.is_password=is_password #must be initialized before text definition (in super())
+    def __init__(self, text, pos, size, color=c.white, is_draggable=True, relative_pos=[0, 0],type='text',selection_color=c.red, file_types=None):
```
Comment (DovaX) [inline pgwidget/pgwidget_core.py:1424]: @Omastto1 Breaking change - be careful about these - but probably meaningful - could you add docstring with explanation/examples + change "type" parameter to Literal/Enum so we know what options are available.

File: pgwidget/pgwidget_core.py [hunk 2]
```diff
@@ -1420,9 +1421,12 @@ def on_unclick(self, glc):
...
+        self.type=type #must be initialized before text definition (in super())
```
Comment (DovaX) [inline pgwidget/pgwidget_core.py]: did you check whether "is?password" is not used in the predecessor - according to the comment here?

File: pgwidget/pgwidget_core.py [hunk 3]
```diff
@@ -1420,9 +1421,12 @@ def on_unclick(self, glc):
...
         super().__init__(text, pos, size, color=color, is_draggable=is_draggable, relative_pos=relative_pos,selection_color=selection_color)
+        if file_types is None:
+            file_types = []
+        self.file_types = file_types
```
Comment (DovaX) [inline pgwidget/pgwidget_core.py:1442]: maybe rename to input_file_types + add docstring, default behaviour could accept any file type (maybe its working like that with empty list as you have it, I dont remember)

**Review by DovaX — CHANGES_REQUESTED** [review]

File: pgwidget/pgwidget_core.py (hunk 3 above)
Comment (Omastto1) [reply pgwidget/pgwidget_core.py:1442]: i expect we can use this for outputting files as well. keeping file_types name

Comment (Omastto1) [reply pgwidget/pgwidget_core.py:1442]: replaced `[]` with explicit [('all files', '*')]

**Review by DovaX — APPROVED** [review]

**Pull request merged**